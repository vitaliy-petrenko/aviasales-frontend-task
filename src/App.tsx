import React from 'react'
import DoNotTouchThis from './pages/DoNotTouchThis'
import FindTickets from './pages/FindTickets'

import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom'

const App: React.FC = () => (
  <Router>
    <Switch>
      <Route exact path='/'>
        <Redirect to='/find-tickets'/>
      </Route>
      <Route exact path='/find-tickets'>
        <FindTickets/>
      </Route>
      <Route exact path='/o-privet'>
        <DoNotTouchThis/>
      </Route>
      <Route path='*'>
        <Redirect to='/o-privet'/>
      </Route>
    </Switch>
  </Router>
)

export default App
