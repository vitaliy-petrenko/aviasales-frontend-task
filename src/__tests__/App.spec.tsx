import React from 'react'
import App from '../App'
import { shallow } from 'enzyme'

describe('App', () => {
  const component = shallow(<App/>)

  it('renders properly', () => {
    expect(component).toMatchSnapshot()
  })
})
