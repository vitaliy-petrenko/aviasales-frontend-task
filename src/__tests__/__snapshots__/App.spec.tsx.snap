// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<HashRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Redirect
        to="/find-tickets"
      />
    </Route>
    <Route
      exact={true}
      path="/find-tickets"
    >
      <FindTickets />
    </Route>
    <Route
      exact={true}
      path="/o-privet"
    >
      <WTF />
    </Route>
    <Route
      path="*"
    >
      <Redirect
        to="/o-privet"
      />
    </Route>
  </Switch>
</HashRouter>
`;
